subprojects/packagefiles/openssl/meson.build

```diff
@@ -0,0 +1,482 @@
+    arch_subdir = 'linux-ppc64le'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'ppc64' and host_machine.system() == 'linux'
+    arch_subdir = 'linux-ppc64'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 's390x' and host_machine.system() == 'linux'
+    arch_subdir = 'linux64-s390x'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'arm' and host_machine.system() in ['linux', 'android']
+    arch_subdir = 'linux-armv4'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'aarch64' and host_machine.system() in ['linux', 'android']
+    arch_subdir = 'linux-aarch64'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() in ['dragonfly', 'freebsd', 'netbsd', 'openbsd']
+    arch_subdir = 'BSD-x86'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() in ['linux', 'android']
+    arch_subdir = 'linux-elf'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'darwin'
+    arch_subdir = 'darwin-i386-cc'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'sunos'
+    arch_subdir = 'solaris-x86-gcc'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86' and host_machine.system() == 'windows'
+    arch_subdir = 'VC-WIN32'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() in ['dragonfly', 'freebsd', 'netbsd', 'openbsd']
+    arch_subdir = 'BSD-x86_64'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'darwin'
+    arch_subdir = 'darwin64-x86_64-cc'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'aarch64' and host_machine.system() == 'darwin'
+    arch_subdir = 'darwin64-arm64-cc'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'sunos'
+    arch_subdir = 'solaris64-x86_64-gcc'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'windows'
+    arch_subdir = 'VC-WIN64A'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'aarch64' and host_machine.system() == 'windows'
+    arch_subdir = 'VC-WIN64-ARM'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'x86_64' and host_machine.system() == 'linux'
+    arch_subdir = 'linux-x86_64'
+    asm = 'no-asm'
+  elif host_machine.cpu_family() == 'mips64' and host_machine.system() == 'linux'
+    arch_subdir = 'linux64-mips64'
+    asm = 'no-asm'
+  else
+    error('Unsupported arch+OS combo: ' + host_machine.cpu_family() + ' + ' + host_machine.system())
+  endif
+
+  message('OpenSSL is configured without ASM support')
+else
+  message('OpenSSL is configured with ASM support')
+endif
+
+subdir('generated-config/archs' / arch_subdir / asm)
+
+# Build options specific to OS
+if host_machine.system() == 'windows'
+  defines += [
+    ## default of Win. See INSTALL in openssl repo.
+    'OPENSSLDIR="C:\\Program Files\\Common Files\\SSL"',
```

Comment:
Shouldn't that be based on `get_option('prefix')` ?

Reply:
I believe this is where OpenSSL looks for ca-certificates, so it's fine. In GStreamer we have [a patch that loads it from the prefix](https://gitlab.freedesktop.org/gstreamer/cerbero/-/blob/0217c0b6793ff1430e261938d2764916b3b6d019/recipes/openssl/0001-Load-ca-certificate.crt-from-PREFIX-etc-ssl-on-macOS.patch) (relative to the current libdir), but that's out of scope.